httpbin/httpbin.go

```diff
@@ -179,3 +185,38 @@ func (h *HTTPBin) Handler() http.Handler {
 
 	return handler
 }
+
+func (h *HTTPBin) HeadersProcessor(headers http.Header) http.Header {
+	result := make(http.Header)
+	for k, v := range headers {
+		lowerCase := strings.ToLower(k)
+		skipped := false
+		for _, pattern := range h.excludeHeadersPatterns {
+			matched := pattern.Match([]byte(lowerCase))
+			if matched {
+				skipped = true
+				break
+			}
+		}
+		if skipped {
+			continue
+		}
+		result[k] = v
+	}
+
+	return result
+}
+
+func (h *HTTPBin) SetExcludeHeaders(excludeHeaders string) {
```

Comment:
This doesn't need to be part of the public API.  Let's inline this into the `WithExcludeHeaders` option func below (sort of like we inline the similar logic in `WithAllowedRedirects`).